graph/coloring/graph_test.go

```diff
@@ -0,0 +1,150 @@
+package coloring_test
+
+import (
+	"github.com/TheAlgorithms/Go/graph/coloring"
+	"strconv"
+	"testing"
+)
+
+type testGraph struct {
+	Graph        *coloring.Graph
+	ColorsUsed   int
+	VertexColors map[int]coloring.Color
+}
+
+func getTestGraphs() (list []*testGraph) {
+	// Graph 0th:
+	// 		1---2
+	// 		| /  \
+	//  4---3     0
+	// Min number of colors required = 3
+	g0 := &testGraph{
+		Graph:      &coloring.Graph{},
+		ColorsUsed: 3,
+		VertexColors: map[int]coloring.Color{
+			1: 1, 4: 1, 0: 1,
+			2: 2,
+			3: 3,
+		},
+	}
+	list = append(list, g0)
+	g0.Graph.AddEdge(4, 3)
+	g0.Graph.AddEdge(3, 1)
+	g0.Graph.AddEdge(3, 2)
+	g0.Graph.AddEdge(1, 2)
+	g0.Graph.AddEdge(2, 0)
+
+	// Graph 1st:
+	// 		1---2
+	// 		| / |
+	//  4---3---0
```

Comment:
Please here and in other places use spaces instead of tabs because the ascii pictures are distorted because of the latter

Reply:
Yeah, sorry. I always forgot to use spaces instead of tabs in diagrams.